mbedtls/src/ssl/ticket.rs

```diff
@@ -20,46 +28,58 @@ pub trait TicketCallback {
         end: *const c_uchar,
         tlen: *mut size_t,
         lifetime: *mut u32,
-    ) -> c_int;
+    ) -> c_int where Self: Sized;
     unsafe extern "C" fn call_parse(
         p_ticket: *mut c_void,
         session: *mut ssl_session,
         buf: *mut c_uchar,
         len: size_t,
-    ) -> c_int;
+    ) -> c_int where Self: Sized;
 
-    fn data_ptr(&mut self) -> *mut c_void;
+    fn data_ptr(&self) -> *mut c_void;
 }
 
+
 define!(
-    #[c_ty(ssl_ticket_context)]
-    struct TicketContext<'rng>;
-    const init: fn() -> Self = ssl_ticket_init;
+    #[c_box_ty(ssl_ticket_context)]
```

Comment:
Please point to a specific reason boxing is needed here

Reply:
reverted to non-box.